framework/PostProcessorFunctions/HStoPSoperator.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on Feb 02, 2018
+@author: alfoa
+"""
+
+#for future compatibility with Python 3--------------------------------------------------------------
+from __future__ import division, print_function, unicode_literals, absolute_import
+import warnings
+warnings.simplefilter('default',DeprecationWarning)
+if not 'xrange' in dir(__builtins__):
+  xrange = range
+#End compatibility block for Python 3----------------------------------------------------------------
+
+#External Modules------------------------------------------------------------------------------------
+import os
+import copy
+import numpy as np
+#External Modules End--------------------------------------------------------------------------------
+
+from PostProcessorInterfaceBaseClass import PostProcessorInterfaceBase
+
+
+class HStoPSoperator(PostProcessorInterfaceBase):
+  """
+   This Post-Processor performs the conversion from HistorySet to PointSet
+   The conversion is performed based on any of the following operations:
+   - row value
+   - pivot value
+   - operator
+  """
+
+  def initialize(self):
+    """
+     Method to initialize the Interfaced Post-processor
+     @ In, None
+     @ Out, None
+    """
+    PostProcessorInterfaceBase.initialize(self)
+    self.inputFormat  = 'HistorySet'
+    self.outputFormat = 'PointSet'
+    #pivotParameter identify the ID of the temporal variable in the data set based on which
+    # the operations are performed. Optional (defaul=time)
+    self.pivotParameter = 'time'
+    self.settings       = {'operationType':None,'operationValue':None,'pivotStrategy':'nearest'}
+
+  def readMoreXML(self,xmlNode):
```

Comment:
I think in the future, we need to use the InputData for parsing the XML in the interface pp.

Reply:
yes. it is a general think we will address in the "new devel"